kube/job_test.go

```diff
@@ -130,3 +130,55 @@ func TestJobPostFlight(t *testing.T) {
 	}
 	testhelpers.IsYAMLSubset(assert, expected, actual)
 }
+
+func TestJobWithAnnotations(t *testing.T) {
+	assert := assert.New(t)
+
+	role := jobTestLoadRole(assert, "role", "job-with-annotation.yml")
+	if role == nil {
+		return
+	}
+
+	job, err := NewJob(role, ExportSettings{
+		Opinions: model.NewEmptyOpinions(),
+	})
+	if !assert.NoError(err, "Failed to create job from role pre-role") {
+		return
+	}
+	assert.NotNil(job)
+
+	yamlConfig := &bytes.Buffer{}
+	if err := helm.NewEncoder(yamlConfig).Encode(job); !assert.NoError(err) {
+		return
+	}
+
+	var expected, actual interface{}
+	if !assert.NoError(yaml.Unmarshal(yamlConfig.Bytes(), &actual)) {
+		return
+	}
+	expectedYAML := strings.Replace(`---
+		apiVersion: extensions/v1beta1
+		kind: Job
+		metadata:
+			name: role
+			annotations:
+				helm.sh/hook: post-install
+	`, "\t", "    ", -1)
+	if !assert.NoError(yaml.Unmarshal([]byte(expectedYAML), &expected)) {
+		return
+	}
+
+	/*
```

Comment:
What's with this chunk of commented out stuff (that we don't actually care about right now)?

Reply:
Ugh, leftover from the structure I copied. :frowning: 

Fixed in bffb7e7.